### PR TITLE
Combining UpdateDependency test for all package managers 

### PR DESCRIPTION
**Pull request by eranturgeman**
…python, nuget)

- [x] All [tests](https://github.com/jfrog/frogbot#tests) passed. If this feature is not already covered by the tests, I added new tests.
- [x] This pull request is on the dev branch.
- [x] I used gofmt for formatting the code before submitting the pull request.
---

Combining all separate updateDependency tests of all package managers into a single test

### CODE REVIEW

File: packagehandlers/commonpackagehandler.go [hunk 1]
```diff
@@ -30,6 +30,8 @@ func GetCompatiblePackageHandler(vulnDetails *utils.VulnerabilityDetails, detail
 		handler = &PythonPackageHandler{pipRequirementsFile: details.PipRequirementsFile}
 	case coreutils.Maven:
 		handler = &MavenPackageHandler{depsRepo: details.DepsRepo, ServerDetails: details.ServerDetails}
+	case coreutils.Nuget, coreutils.Dotnet:
```
Comment (EyalDelarea) [inline packagehandlers/commonpackagehandler.go:33]: This can be removed and added in the next PR, just in case your for some reason merge this one and not the second.

File: packagehandlers/packagehandlers_test.go [hunk 2]
```diff
@@ -34,123 +31,191 @@ type pipPackageRegexTest struct {
...
+				uniqueChecksExtraArgs: []string{"go.mod"},
+			},
+			{
+				vulnDetails: &utils.VulnerabilityDetails{
+					SuggestedFixedVersion:       "1.7.7",
+					IsDirectDependency:          true,
+					VulnerabilityOrViolationRow: formats.VulnerabilityOrViolationRow{Technology: coreutils.Go, ImpactedDependencyName: "github.com/gin-gonic/gin"},
+				},
+				fixSupported:          true,
+				uniqueChecksExtraArgs: []string{"go.mod"},
```
Comment (EyalDelarea) [inline packagehandlers/packagehandlers_test.go:54]: uniqueChecksExtraArgs : Why is this needed if we only use in go to set the go.mod in golang?

File: packagehandlers/packagehandlers_test.go [hunk 3]
```diff
@@ -34,123 +31,191 @@ type pipPackageRegexTest struct {
...
-				SuggestedFixedVersion:       "1.7.7",
-				IsDirectDependency:          true,
-				VulnerabilityOrViolationRow: formats.VulnerabilityOrViolationRow{Technology: coreutils.Go, ImpactedDependencyName: "github.com/gin-gonic/gin"},
+			{
+				vulnDetails: &utils.VulnerabilityDetails{
+					SuggestedFixedVersion:       "1.25.9",
+					VulnerabilityOrViolationRow: formats.VulnerabilityOrViolationRow{Technology: coreutils.Pip, ImpactedDependencyName: "urllib3"},
+				},
+				scanDetails:  &utils.ScanDetails{Project: &utils.Project{PipRequirementsFile: "requirements.txt"}},
+				fixSupported: false,
```
Comment (EyalDelarea) [inline packagehandlers/packagehandlers_test.go:75]: Just make sure about `FixSupported` bool, it is intended mostly for direct vs indirects fixes compatibility
i'm not sure `urllib3` is indirect in this case, just need to make sure.

For example if i'm getting indirect dependency in NPM i want to make sure we don't attempt to fix it, until we have the proper implementation

**Review by EyalDelarea — CHANGES_REQUESTED** [review]

File: packagehandlers/packagehandlers_test.go (hunk 2 above)
Comment (eranturgeman) [reply packagehandlers/packagehandlers_test.go:54]: because the test function is unifying all package managers I wanted to make it general as much as possible so I dont want to add in the general function spacial case for Go.
also when I implement it like this it makes it more sense and orginized for future addition for other package managers without overloading the main function which is long anyway

File: packagehandlers/packagehandlers_test.go (hunk 3 above)
Comment (eranturgeman) [reply packagehandlers/packagehandlers_test.go:75]: I just moved the python tests into a unified function, but as for the logic the test must get a dependencyFixTest structure in order to work, but in this case (same as other package managers) we define a dependency and define it manually as indirect to see if the flow drops. here we simulate a dependency we get which is indirect and therefore the fix should not be supported
